update registering an account (#345)
hyperledger/iroha-2-docs: src/guide/kotlin-java.md

We assume that you know how to create a new package and have basic
understanding of the fundamental Kotlin code. Specifically, we shall assume
that you know how to build and deploy your program on the target platforms.

> To clone Iroha 2 JVM compatible SDKs, you can use https://github.com/hyperledger/iroha-java.

Reviewer: ```suggestion
To clone Iroha 2 JVM compatible SDKs, you can use [Iroha Java](https://github.com/hyperledger/iroha-java).
```

<<<@/snippets/IrohaConfig.kotlin

> ## 3. Domain search and registration

Reviewer 2: ```suggestion
## 3. Querying and Registering Domains
```

> Search Registering a domain is one of the easier operations. The usual boilerplate

Reviewer: search registered domains?

Author: Search and registering - sure

Reviewer 2: ```suggestion
Querying and Registering a domain are among the easier operations. The usual boilerplate
```

> Querying and Registering a domain are among the easier operations. The usual boilerplate

Reviewer 3: ```suggestion
Querying and Registering a domain are easier operations. The usual boilerplate
```

terminology), hence blocking is not necessarily the only mode of
interaction with the Iroha 2 code.

> In order to make sure that the raised peers work correctly, you can do a simple 

Reviewer: running peers maybe?

Author: oh, thank you, you are right

    val adminKeyPair = keyPairFromHex("7233bfc89dcbd68c19fde6ce6158225298ec1131b6a130d1aeb454c1ab5183c0",
        "9ac47abf59b356e0bd7dcbbbb4dec080e302156a48ca907e47cb6aea1d32719e")

    val client = AdminIroha2Client(URL(peerUrl), URL(telemetryUrl), log = true)
    val query = Query(peerUrl, telemetryUrl, client, admin, adminKeyPair)

    query.findAllDomains()
        .also { println("ALL DOMAINS: ${it.map { d -> d.id.asString() }}") }

}

open class Query (peerUrl: String,
                  telemetryUrl: String,
                  private val client: AdminIroha2Client,
                  private val admin: AccountId,
                  private val keyPair: KeyPair) {
    
    suspend fun findAllDomains(queryFilter: GenericValuePredicateBox<ValuePredicate>? = null) = QueryBuilder
        .findAllDomains(queryFilter)
        .account(admin)
        .buildSigned(keyPair)
        .let { client.sendQuery(it) }
}
```

The output in the terminal will contain a list of all domains that are currently registered

::: details Expand to see the expected output

```
ALL DOMAINS: [wonderland, genesis, garden_of_live_flowers]
```

:::

> To register a new domain add next lines on Main.kt

Reviewer: ```suggestion
To register a new domain, add the following lines to Main.kt:
```

```kotlin
val sendTransaction = SendTransaction(peerUrl, telemetryUrl, client, admin, adminKeyPair)

val domain = "domain_${System.currentTimeMillis()}"
    sendTransaction.registerDomain(domain).also { println("DOMAIN $domain CREATED") }
```

> And create new open class SendTransaction to your project.

Reviewer: ```suggestion
Then create new open class `SendTransaction` in your project:
```

When registering an account, there are a few more variables. The account
can only be registered to an existing domain. Also, an account typically
has to have a key pair. 

> To register a new account add next lines on Main.kt

Reviewer: ```suggestion
To register a new account, add the following lines to `Main.kt`:
```

```Kotlin
    val joe = "joe_${System.currentTimeMillis()}$ACCOUNT_ID_DELIMITER$domain"
    val joeKeyPair = generateKeyPair()
    sendTransaction.registerAccount(joe, listOf(joeKeyPair.public.toIrohaPublicKey()))
        .also { println("ACCOUNT $joe CREATED") }

    query.findAllAccounts()
        .also { println("ALL ACCOUNTS: ${it.map { a -> a.id.asString() }}") }
```

> And implement new method on class SendTransaction to your project.

Reviewer: ```suggestion
Then implement new method for class `SendTransaction` in your project.
```
